src/Functions/FunctionDateOrDateTimeToDateTimeOrDateTime64.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+#include <Functions/IFunctionDateOrDateTime.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+}
+
+template <typename Transform>
+class FunctionDateOrDateTimeToDateTimeOrDateTime64 : public IFunctionDateOrDateTime<Transform>, WithContext
```

Comment:
Minor: I don't think it is necessary to inherit from `WithContext` (same applies to "FunctionDateOrDateTimeToDateOrDate32.h")

Reply:
It's not minor. You are right: `WithContext` is not needed. I removed it from both places.